enclave-tls/samples/enclave-tls-client/ra-tls-client.c

```diff
@@ -0,0 +1,67 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <errno.h>
+
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+#include <enclave-tls/api.h>
+
+int ra_tls_echo(int sockfd, enclave_tls_log_level_t log_level,
+		char *attester_type, char *verifier_type, char *tls_type)
+{
+	enclave_tls_err_t ret;
+	enclave_tls_handle handle;
+	enclave_tls_conf_t conf;
+
+	memset(&conf, 0, sizeof(conf));
+	conf.fd = sockfd;
```

Comment:
fd should be passed to negotiate, not conf. Please remove fd from conf structure.